analysis/topology/topological_space.lean

```diff
@@ -635,6 +645,71 @@ lemma tendsto_nhds_unique [t2_space α] {f : β → α} {l : filter β} {a b : 
   (hl : l ≠ ⊥) (ha : tendsto f l (nhds a)) (hb : tendsto f l (nhds b)) : a = b :=
 eq_of_nhds_neq_bot $ neq_bot_of_le_neq_bot (map_ne_bot hl) $ le_inf ha hb
 
+def topological_space.separated (s t : set α) : Prop :=
+∃u v : set α, is_open u ∧ is_open v ∧ s ⊆ u ∧ t ⊆ v ∧ u ∩ v = ∅
+
+open topological_space
+
+lemma separated_swap {s t : set α} : separated s t → separated t s :=
+assume ⟨u, v, uo, vo, su, tv, uv⟩, ⟨v, u, vo, uo, tv, su, inter_comm u v ▸ uv⟩
+
+lemma separated_symm {s t : set α} : separated s t ↔ separated t s :=
+iff.intro separated_swap separated_swap
```

Comment:
I used to be confused about the difference between `comm` and `symm`, but now I think I've got it worked out. `comm` is for statements whose conclusion is bidirectional: an equality or, for propositions, an iff. `symm` is for statements whose conclusion is unidirectional: a function or implication. Field notation is useful with `symm`.

(In math, we tend to say a "symmetric relation" but a "commutative operation", and we don't tend to distinguish much between "for all a and b, a ~ b implies b ~ a" and "for all a and b, a ~ b iff b ~ a", for obvious reasons! so the convention is a bit perplexing at first.)

I've renamed `separated` to `separated_sets` because the `open topological_space` that I was using didn't get along well with using field notation later on. The purpose of that was to avoid a name collision with `separated` for uniform spaces.